File: ts/day01p1/main.py
#!/usr/bin/env python3

import sys

fname = sys.argv[1] if len(sys.argv[1:]) == 1 else "input.txt"

# # stupid single expression
# print(
#     sum(
#         abs(a - b)
#         for a, b in zip(
#             *map(
#                 sorted,
#                 zip(*[map(int, line.strip().split()) for line in open(fname)]),
#             )
#         )
#     )
# )

left = []
right = []
for line in open(fname):
    a, b = map(int, line.strip().split())
    left.append(a)
    right.append(b)

left.sort()
right.sort()

delta = sum(abs(a - b) for a, b in zip(left, right))

print(delta)

File: ts/day01p1/main.ts
#!/usr/bin/env -S deno -A

export function historianHysteria(lines: string[]): number {
  let runningDelta = 0;
  const left: number[] = [];
  const right: number[] = [];

  for (let line of lines) {
    line = line.trim();
    if (line.length == 0) {
      console.log("SKIPPING");
      continue;
    }
    const [a, b] = line.split(/\s+/).map(Number);
    left.push(a);
    right.push(b);
  }

  left.sort();
  right.sort();

  for (let i = 0; i < left.length; i++) {
    const a = left[i];
    const b = right[i];
    const delta = Math.abs(a - b);
    console.log(a, b, delta);
    runningDelta += delta;
  }
  return runningDelta;
}

// Learn more at https://docs.deno.com/runtime/manual/examples/module_metadata#concepts
if (import.meta.main) {
  const text = (await Deno.readTextFile("input.txt")).trimEnd();
  console.log(historianHysteria(text.split("\n")));
}
